ion/gen.c

```diff
@@ -40,7 +40,7 @@ const char *gen_preamble_str =
     "#define alignof(x) __alignof__(x)\n"
     "#endif\n"
     "\n"
-    "#define va_start_ptr(args, arg) (va_start(*(args), *(arg)))\n"
+    "#define va_start_ptr(args, arg) (va_start(*(args), arg))\n"
```

Comment:
The existing behavior is intentional, if a little odd to work around the fact that va_start in stdarg.h officially takes a parameter name as the argument but will accept an lvalue on all the compilers/platforms I tested. I think the correct thing to do is to disable the warning, and just have tests that verify the expected behavior.

Reply:
Sounds good. I'll test with a surrounding pragma and if I can figure out where to best put a behavior test I'll try to add that as well.

Reply:
If you add the pragmas and make sure it fixes the warnings on your end, that'd be fine for the PR. I can add the tests myself.